crates/sui-types/src/base_types.rs

```diff
@@ -1017,6 +1017,10 @@ pub struct TxContext {
     epoch_timestamp_ms: CheckpointTimestamp,
     /// Number of `ObjectID`'s generated during execution of the current transaction
     ids_created: u64,
+    // gas price passed to transaction as input
+    gas_price: u64,
```

Comment:
What about budget?

Reply:
I thought from previous discussions budget was what we wanted exposed, not gas price?

Reply:
I think it's possible we will not expose either.
I wanted to put one of the two here so we get a good sense of how to pass the info if needed.
So I was planning to clean up price/budget on a later PR according to the conclusion on the conversation going.
We could have done without either price or budget, but as I said it felt good to show how things would be done

Reply:
in other words passing `GasData` gives us access to everything we may possibly need and that is what this PR is all about